src/System.ObjectModel/src/System/Collections/ObjectModel/ObservableCollection.cs

```diff
@@ -46,6 +46,22 @@ public class ObservableCollection<T> : Collection<T>, INotifyCollectionChanged,
         /// <exception cref="ArgumentNullException"> collection is a null reference </exception>
         public ObservableCollection(IEnumerable<T> collection) : base(CreateCopy(collection)) { }
 
+        /// <summary>
+        /// Initializes a new instance of the ObservableCollection class
+        /// that contains elements copied from the specified list
+        /// </summary>
+        /// <param name="list">The list whose elements are copied to the new list.</param>
+        /// <remarks>
+        /// The elements are copied onto the ObservableCollection in the
+        /// same order they are read by the enumerator of the list.
+        /// </remarks>
+        /// <exception cref="ArgumentNullException"> list is a null reference </exception>
+        public ObservableCollection(List<T> list)
+            : base((list != null) ? new List<T>(list.Count) : list)
+        {
+            CreateCopy(list);
+        }
```

Comment:
Or, since `List<T>`'s constructor already throws `ArgumentNullException` with a `paramName` of `collection`, it could be simplified by modifying `CreateCopy` to take `List<T>` param named `list`:

```c#
public ObservableCollection(IEnumerable<T> collection) : base(new List<T>(collection)) { }

public ObservableCollection(List<T> list) : base(CreateCopy(list)) { }

private static List<T> CreateCopy(List<T> list)
{
    if (list == null)
        throw new ArgumentNullException(nameof(list));

    return new List<T>(list);
}
```